operator/controllers/seldondeployment_controller.go

```diff
@@ -81,13 +82,29 @@ type components struct {
 	hpas             []*autoscaling.HorizontalPodAutoscaler
 	virtualServices  []*istio.VirtualService
 	destinationRules []*istio.DestinationRule
+	addressable      *machinelearningv1.SeldonAddressable
 }
 
 type httpGrpcPorts struct {
 	httpPort int
 	grpcPort int
 }
 
+func createAddressableResource(mlDep *machinelearningv1.SeldonDeployment, namespace string) (*machinelearningv1.SeldonAddressable, error) {
+	// It was an explicit design decision to expose the service name instead of the ingress
+	// Currently there will only be a URL for the first predictor, and assumes always REST
+	firstPredictorName := mlDep.Spec.Predictors[0].Name
+	addressablePort, err := getEngineHttpPort()
+	if err != nil {
+		return nil, err
+	}
+	addressableHost := mlDep.Name + "-" + firstPredictorName + "." + namespace + ".svc.cluster.local" + ":" + strconv.Itoa(addressablePort)
```

Comment:
is this logic for svc name not in a method. Seems this could break if we don't use that method and things change in future?

Reply:
Good point, I've addressed it